drivers/spi/CMakeLists.txt

```diff
@@ -2,7 +2,8 @@ zephyr_library()
 
 zephyr_library_sources_ifdef(CONFIG_SPI_DW		spi_dw.c)
 zephyr_library_sources_ifdef(CONFIG_SPI_INTEL		spi_intel.c)
-zephyr_library_sources_ifdef(CONFIG_SPI_STM32		spi_ll_stm32.c)
+zephyr_library_sources_codegen_ifdef(CONFIG_SPI_STM32		spi_ll_stm32.c)
```

Comment:
@carlescufi , @SebastianBoe : Heads up: This is one point where codegen depends on Kconfig.
Here codegen is called based on CONFIG_xxx flag.

Might not be an issue if there are multiple steps:
device tree > dts dtb > Kconfig Symbols generation > Codegen for C code generation